tests/e2e/contracts/multi_tenant_storage.py

```
@@ -0,0 +1,48 @@
+    def get_all_storages(self) -> dict[str, str]:
+        return {
+            storage_contract.as_hex: gl.ContractAt(storage_contract)
+            .view()
+            .get_storage()
+            for storage_contract in self.all_storage_contracts
+        }
+
+    @gl.public.write
+    def update_storage(self, new_storage: str) -> None:
+        # Assign user to a storage contract if not already assigned
+        if gl.message.sender_address not in self.mappings:
```

Ruff (0.11.9)

`gl` may be undefined, or defined from star imports

(F405)